### PR TITLE
Refactor: SwingBuilder API

### PR DESCRIPTION
**Pull request by DanVanAtta**
(1) Do a long standing cleanup to use constructor instead of 'builder()' factory method. EG:
Instead of "SwingBuilder.builder()", use "new SwingBuilder()"

(2) Move JPanelBuilder to a 'jpanel' package in prep for layout helper classes to land in the same package

(3) Move layout bulder API calls to be last in builder chain in prep for type-safe builders. EG:

Instead of:
```
new JPanelBuilder()
     .borderLayout()
     .border(...)
     .build();
```
Change to:
```
new JPanelBuilder()
     .border(...)
     .borderLayout()
     .build();
```

In the latter case, when there are type-safe builders, the layout builder will no longer have access to the 'border' method and it will need to then appear before any layout builders are invoked.

 

## Functional Changes

[ ] New map or map update
[ ] New Feature
[ ] Feature update or enhancement
[ ] Feature Removal
[x] Code Cleanup or refactor
[ ] Configuration Change
[ ] Bug fix:  
[ ] Other:   

## Testing

[x] Covered by existing automated tests
[ ] Covered by newly added automated tests
[ ] Manually tested
[x] No testing done

### CODE REVIEW

File: game-core/src/main/java/games/strategy/engine/framework/startup/ui/panels/main/MainPanel.java [hunk 1]
```diff
@@ -38,20 +38,19 @@
...
-  private final JButton cancelButton = JButtonBuilder.builder().title("Cancel").build();
+  private final JButton cancelButton = new JButtonBuilder().title("Cancel").build();
 
-  private final JPanel gameSetupPanelHolder = JPanelBuilder.builder().borderLayout().build();
+  private final JPanel gameSetupPanelHolder = new JPanelBuilder().borderLayout().build();
   private final JPanel mainPanel;
   private final JSplitPane chatSplit;
-  private final JPanel chatPanelHolder =
-      JPanelBuilder.builder().borderLayout().preferredHeight(62).build();
+  private final JPanel chatPanelHolder = new JPanelBuilder().preferredHeight(62).build();
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/framework/startup/ui/panels/main/MainPanel.java]: What happened to borderLayout?

File: game-core/src/main/java/games/strategy/engine/lobby/moderator/toolbox/tabs/bad/words/BadWordsTab.java [hunk 2]
```diff
@@ -113,10 +115,9 @@ private JPanel buildAddBadWordsPanel(final JTable table) {
     DocumentListenerBuilder.attachDocumentListener(
         addField, () -> addButton.setEnabled(addField.getText().trim().length() >= MIN_LENGTH));
 
-    return JPanelBuilder.builder()
+    return new JPanelBuilder()
         .add(
-            JPanelBuilder.builder()
-                .flowLayout()
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/lobby/moderator/toolbox/tabs/bad/words/BadWordsTab.java]: FlowLayout?

File: swing-lib/src/main/java/org/triplea/swing/JButtonBuilder.java [hunk 3]
```diff
@@ -32,11 +32,7 @@ public JButtonBuilder(final String title) {
     this.title = title;
   }
 
-  private JButtonBuilder() {}
-
-  public static JButtonBuilder builder() {
-    return new JButtonBuilder();
-  }
+  public JButtonBuilder() {}
```
Comment (RoiEXLab) [inline swing-lib/src/main/java/org/triplea/swing/JButtonBuilder.java]: Unless there's another constructor in this class having a no-op public constructor is redundant

File: swing-lib/src/main/java/org/triplea/swing/jpanel/JPanelBuilder.java [hunk 4]
```diff
@@ -51,7 +51,7 @@
   private int gridBagHelperColumns;
   private Integer preferredHeight;
 
-  private JPanelBuilder() {}
+  public JPanelBuilder() {}
```
Comment (RoiEXLab) [inline swing-lib/src/main/java/org/triplea/swing/jpanel/JPanelBuilder.java]: Same here with the constructor

File: game-core/src/main/java/games/strategy/engine/lobby/moderator/toolbox/tabs/bad/words/BadWordsTab.java (hunk 2 above)
Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/engine/lobby/moderator/toolbox/tabs/bad/words/BadWordsTab.java]: flow layout is/becomes the default

File: game-core/src/main/java/games/strategy/engine/framework/startup/ui/panels/main/MainPanel.java (hunk 1 above)
Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/engine/framework/startup/ui/panels/main/MainPanel.java]: Looks like it just got dropped. Added back in: 3aaff4467

File: swing-lib/src/main/java/org/triplea/swing/JButtonBuilder.java (hunk 3 above)
Comment (DanVanAtta) [reply swing-lib/src/main/java/org/triplea/swing/JButtonBuilder.java]: There is another constructor above. The position of this constructor is not terribly standard though, usually no-arg ones come before.

Updated ordering: cfac03cf4

**Pull request merged**